test/propagate_constant_test.cpp

```diff
@@ -163,4 +164,60 @@ TEST_CASE(const_dot)
     EXPECT(m1 == m2);
 }
 
+TEST_CASE(literal_convert_scalar)
+{
+    migraphx::module m;
+
+    auto lit = m.add_literal(
+        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {1}, {0}}));
+    auto t = m.add_instruction(
+        migraphx::make_op("convert",
+                          {{"target_type", migraphx::to_value(migraphx::shape::int32_type)}}),
+        lit);
+    m.add_return({t});
+
+    auto out_shape = m.get_output_shapes().back();
+    run_pass(m);
+
+    auto target_lit_out_shape = migraphx::shape{migraphx::shape::int32_type, {1}, {0}};
+    EXPECT(target_lit_out_shape == out_shape);
+}
+
+TEST_CASE(literal_convert_vec)
+{
+    migraphx::module m;
+
+    auto lit = m.add_literal(
+        migraphx::generate_literal(migraphx::shape{migraphx::shape::float_type, {3, 3, 3}}));
+    auto t = m.add_instruction(
+        migraphx::make_op("convert",
+                          {{"target_type", migraphx::to_value(migraphx::shape::int32_type)}}),
+        lit);
+    m.add_return({t});
+
+    auto out_shape = m.get_output_shapes().back();
+    run_pass(m);
+
+    auto target_lit_out_shape = migraphx::shape{migraphx::shape::int32_type, {3, 3, 3}};
+    EXPECT(target_lit_out_shape == out_shape);
+}
+
+TEST_CASE(literal_gather_propagate)
+{
+    migraphx::module m;
+
+    auto lit = m.add_literal(
+        migraphx::literal{migraphx::shape(migraphx::shape::int32_type, {3}), {3, 800, 800}});
+    auto lit2 = m.add_literal(
+        migraphx::literal{migraphx::shape(migraphx::shape::int32_type, {1}, {0}), {1}});
+    auto t = m.add_instruction(migraphx::make_op("gather"), lit, lit2);
+    m.add_return({t});
+
+    auto out_shape = m.get_output_shapes().back();
+    run_pass(m);
+
+    auto target_lit_out_shape = migraphx::shape{migraphx::shape::int32_type, {1}, {0}};
+    EXPECT(target_lit_out_shape == out_shape);
```

Comment:
`out_shape` would match even if it didn't const-fold.  
It should assert that after running pass on your module, there is no `convert' instructions left.

Reply:
Or you can check for `EXPECT(m1 == m2)` where `m1` is  output of run_pass(m) and m2 is expected module.

Reply:
Done